Acoustic

Suchanekf opened the pull request:
## Description
Provide a brief description of the PR's purpose here.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] TODO 1

## Questions
- [ ] Question1

## Status
- [ ] Ready to go

Suchanekf committed:
materials refractive indexes

Suchanekf committed:
Add files via upload

Suchanekf committed:
Delete materials.py

Suchanekf committed:
Add files via upload

Suchanekf committed:
Delete materials.py

Suchanekf committed:
Add files via upload

Suchanekf committed:
Delete materials.py

Suchanekf committed:
Add files via upload

Suchanekf committed:
Add files via upload

Suchanekf committed:
Add files via upload

Suchanekf committed:
Delete material_references.md

Suchanekf committed:
Add files via upload

gtg724t committed:
added something to references

gtg724t committed:
made a small change to material_references

FoleyLab committed:
Merge pull request #23 from FoleyLab/refs

Refs

Suchanekf committed:
refractive indexes for various materials

Suchanekf committed:
Micrometer is converted to meter

Suchanekf committed:
micrometer is convetred into meters

FoleyLab committed:
Update material_references.md

FoleyLab committed:
Update materials.py

FoleyLab committed:
Update test_materials.py

FoleyLab committed:
Merge pull request #24 from FoleyLab/FS_mat

refractive indexes for various materials

gtg724t committed:
made some small additions to mie stuff

gtg724t committed:
made small changes to simple

FoleyLab committed:
Merge branch 'main' into miecompute

FoleyLab committed:
Merge pull request #25 from FoleyLab/miecompute

Miecompute

gtg724t committed:
fixed unit tests for updated MieDriver class

FoleyLab committed:
Merge pull request #26 from FoleyLab/miecompute

fixed unit tests for updated MieDriver class

varnerj31 committed:
ed matpy

varnerj31 committed:
added Au and Ag to mats with passing tests

varnerj31 committed:
te up poly sty

varnerj31 committed:
Problems with Al2O3 test

Belloj3 committed:
Pushing Pb,Pt,Re

Belloj3 committed:
Merge branch 'main' into materials2

FoleyLab committed:
Update materials.py

FoleyLab committed:
Update materials.py

FoleyLab committed:
Update test_materials.py

FoleyLab committed:
Merge pull request #27 from FoleyLab/materials2

Materials2

lakatosa1 committed:
Rh data

lakatosa1 committed:
Ru.txt

lakatosa1 committed:
Si data

lakatosa1 committed:
Merge branch 'main' of https://github.com/FoleyLab/wpspecdev into AL_mat

lakatosa1 committed:
functions for Ru, Rh, and Si

lakatosa1 committed:
fixing a typo

lakatosa1 committed:
test functions for Rh, Ru, and Si

FoleyLab committed:
Delete Ru.txt

FoleyLab committed:
Update materials.py

FoleyLab committed:
Update test_materials.py

FoleyLab committed:
Update test_materials.py

FoleyLab committed:
Update materials.py

FoleyLab committed:
Merge pull request #28 from FoleyLab/AL_mat

Al mat

gtg724t committed:
made small update to readme

varnerj31 committed:
dded materials with passing tests

varnerj31 committed:
g Au PS and Al2O3 were added with passing tests

lakatosa1 committed:
New Ru data

lakatosa1 committed:
Ru test

FoleyLab committed:
Merge pull request #30 from FoleyLab/AL_mat

Al mat

FoleyLab committed:
Merge branch 'main' into JV_mat

FoleyLab committed:
Merge pull request #29 from FoleyLab/JV_mat

Jv mat

gtg724t committed:
formatted all python files with black, reverted to snake case for test functions

FoleyLab committed:
Update README.md

gtg724t committed:
added first pass at auto-selection of material data sets

gtg724t committed:
small update to readme + multiple data

FoleyLab committed:
Merge branch 'main' into mult_data

FoleyLab committed:
Merge pull request #31 from FoleyLab/mult_data

Mult data

gtg724t committed:
added autoselect to ta2o5 and corresponding unit test

varnerj31 committed:
Merge pull request #32 from FoleyLab/matauto

added autoselect to ta2o5 and corresponding unit test

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Update .codecov.yml

varnerj31 committed:
revised Au data and test, Ta2O5, Al, W and AlN failed tests

FoleyLab committed:
Update README.md

gtg724t committed:
made a small change to readme

varnerj31 committed:
removed misplaced files

FoleyLab committed:
Merge pull request #34 from FoleyLab/JV_data

Jv data

Suchanekf committed:
refractive indexes for two different data sets

FoleyLab committed:
chanted camel case to snake case for test functions

Deleted old test functions that did not test both data sets.

FoleyLab committed:
Merge pull request #35 from FoleyLab/FS_mat

refractive indexes for two different data sets

gtg724t committed:
added a few functions

gtg724t committed:
can compute transfer matrix now

gtg724t committed:
maybe have fixed TMM method, let's see if it passes

gtg724t committed:
tmm is passing for basic case

FoleyLab committed:
Merge pull request #36 from FoleyLab/em_try

gtg724t committed:
first pass at integrating materials class with TmmDriver

gtg724t committed:
updated em.py to integrate materials class and improved testing

FoleyLab committed:
Update material_references.md

gtg724t committed:
added release.yml - will it automatically upload to PyPI?

gtg724t committed:
added release.yml and fixed non-passing tests

FoleyLab committed:
Merge pull request #37 from FoleyLab/mat_integrate

Mat integrate

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

gtg724t committed:
trying to tag this version

Suchanekf committed:
Si revised data sets

Suchanekf committed:
Revised Si data sets

FoleyLab committed:
Update materials.py

FoleyLab committed:
changed test_material_si to snake case and changed lower_wavelength associated with Shkondin data

FoleyLab committed:
Merge pull request #40 from FoleyLab/FSrev_mat

F srev mat

lakatosa1 committed:
Re Windt data set

lakatosa1 committed:
Re Palik data set from wptherml

lakatosa1 committed:
Re data

lakatosa1 committed:
test for Re

lakatosa1 committed:
Fixed Re data

lakatosa1 committed:
changed Re_windt wavelengths

lakatosa1 committed:
change Re_windt data

FoleyLab committed:
changed the test wavelength for Re_Palik

from 0.007508 -> 0.0000007508

FoleyLab committed:
Merge pull request #41 from FoleyLab/AL_mat2

Al mat2

Suchanekf committed:
revised Ag tests

FoleyLab committed:
Merge pull request #43 from FoleyLab/Ag_rev

Suchanekf committed:
added docstrings

FoleyLab committed:
Merge pull request #44 from FoleyLab/FS_DS

added docstrings

gtg724t committed:
added some Mie features plus outline for code to compute g

FoleyLab committed:
Merge pull request #45 from FoleyLab/g_calc

added some Mie features plus outline for code to compute g

gtg724t committed:
added template for numerical gradients of P matrix

gtg724t committed:
added basic therml library

gtg724t committed:
ran black to fix formatting compliance issues with all python files

Suchanekf committed:
Merge pull request #46 from FoleyLab/therml

Therml

Suchanekf committed:
Eq.17 implemented

Suchanekf committed:
revised Equation 17

Suchanekf committed:
made suggested changes

FoleyLab committed:
Merge pull request #48 from FoleyLab/FS_Eq.17

Fs eq.17

Belloj3 committed:
commit

lakatosa1 committed:
equation 27

gtg724t committed:
changed therml methods to use trapz over integral

FoleyLab committed:
Merge pull request #49 from FoleyLab/trapz

Belloj3 committed:
Merge branch 'main' of https://github.com/FoleyLab/wpspecdev into Pb_updated

FoleyLab committed:
removed old material_Pb() method

FoleyLab committed:
deleted duplicate test_material_pb()

FoleyLab committed:
fixed indentation in material_Ag() and material_Pb

FoleyLab committed:
made small change

FoleyLab committed:
Merge pull request #52 from FoleyLab/Pb_updated

Pb updated

gtg724t committed:
fixed error in materials.py

gtg724t committed:
made one change i want to keep and one change I want to disregard in the PR

FoleyLab committed:
Delete README.md

FoleyLab committed:
Merge pull request #54 from FoleyLab/pr_test

made one change i want to keep and one change I want to disregard in …

FoleyLab committed:
Create readme.md

FoleyLab committed:
Update readme.md

Belloj3 committed:
Merge pull request #53 from FoleyLab/pb_update

fixed error in materials.py

FoleyLab committed:
Cancelled nightly CI

You can uncommont the schedule strings to reinitiate nightly CI

lakatosa1 committed:
equation 27

FoleyLab committed:
Merge pull request #56 from FoleyLab/AL_equation

Belloj3 committed:
commit

FoleyLab committed:
Merge pull request #57 from FoleyLab/therml

commit

gtg724t committed:
added unit tests for Eq. 18 and 27

Belloj3 committed:
Merge pull request #58 from FoleyLab/therml_test

added unit tests for Eq. 18 and 27

gtg724t committed:
calculations for qilin

gtg724t committed:
added water as a background medium, deleted some redundant materials

FoleyLab committed:
added pm_gradient function

FoleyLab committed:
Merge pull request #59 from FoleyLab/pm_grad

added pm_gradient function

Suchanekf committed:
added tmm analtical gradient

FoleyLab committed:
Merge pull request #60 from FoleyLab/FS_tmm

added tmm analytical gradient

Suchanekf committed:
_tm_grad is modified

Suchanekf committed:
_tm_grad is revised

Suchanekf committed:
revised test for em.py

FoleyLab committed:
Merge pull request #62 from FoleyLab/_tm_grad

Tm grad

FoleyLab committed:
now test is passing

FoleyLab committed:
added some comments to these files

FoleyLab committed:
added some code to the compute_spectrum_gradient method

FoleyLab committed:
added color rendering feature to em.py

FoleyLab committed:
Merge pull request #63 from FoleyLab/tm_gradient_fix

Tm gradient fix

Suchanekf committed:
transmissivity_gradient_array

Suchanekf committed:
the factor added

FoleyLab committed:
Merge pull request #64 from FoleyLab/FS_SG

transmissivity_gradient_array

FoleyLab committed:
added tests for compute_spectrum_gradient which are passing

FoleyLab committed:
Update readme.md

FoleyLab committed:
now reads AM1.5 data upon instantiation of TMMDriver

FoleyLab committed:
updated the notebook for stpv-pv application

FoleyLab committed:
Merge pull request #65 from FoleyLab/spec_grad_test

added tests for compute_spectrum_gradient which are passing

FoleyLab committed:
added stpv examples

FoleyLab committed:
updated readme

FoleyLab committed:
reupdated readme

FoleyLab committed:
added small example

FoleyLab committed:
added some optimization stuff

FoleyLab committed:
fixed test_env.yaml

FoleyLab committed:
Merge pull request #66 from FoleyLab/opt

fixed test_env.yaml

FoleyLab committed:
added explicit angle capability... needs testing

FoleyLab committed:
added test for explicit angle compute_spectrum

FoleyLab committed:
added cooling functions, need testing

FoleyLab committed:
added cooling functions, 2 of 3 pass unit tests... fix atmospheric radiated power

FoleyLab committed:
fixed atmospheric radiation to agree with old wptherml... double check that cos(theta) should occur as a factor in thermal emission against Fan paper

FoleyLab committed:
added some stpv gradients

FoleyLab committed:
updated doc strings in therml.py

FoleyLab committed:
added explicit angle gradient function - needs testing

FoleyLab committed:
added docs

FoleyLab committed:
Merge pull request #67 from FoleyLab/cooling

Cooling merged

FoleyLab committed:
first draft of readthedocs

FoleyLab committed:
updated document requirements

FoleyLab committed:
updated requirements

FoleyLab committed:
Update requirements.yaml

FoleyLab committed:
finished packaging cooling - needs further testing

FoleyLab committed:
wrapped some functions for radiative cooling

FoleyLab committed:
updated TiN demo

FoleyLab committed:
Merge pull request #68 from FoleyLab/pv-stpv

Pv stpv

FoleyLab committed:
updated examples

FoleyLab committed:
fixed error in cooling gradient functions

FoleyLab committed:
updated documentation of therml.py

FoleyLab committed:
Update readme.md

FoleyLab committed:
updated docs

FoleyLab committed:
Merge branch 'main' of https://github.com/FoleyLab/wpspecdev
updated the badge

FoleyLab committed:
added method to allow users to include new materials by file

FoleyLab committed:
added methods to add and remove layers to multilayer stacks along with unit tests

FoleyLab committed:
updated the cooling demo and some docs

FoleyLab committed:
Update README.md

FoleyLab committed:
Update README.md

FoleyLab committed:
Add files via upload

FoleyLab committed:
Add files via upload

FoleyLab committed:
Update README.md

FoleyLab committed:
started editing stpv demo

FoleyLab committed:
finished draft of STPV Demo

FoleyLab committed:
cleared data in notebook

FoleyLab committed:
testing

FoleyLab committed:
first step towards the new release of wptherml

FoleyLab committed:
prepared new release

FoleyLab committed:
some formatting updates

FoleyLab committed:
tried to delete last wpspecdevs

FoleyLab committed:
updated CI.yaml

FoleyLab committed:
updated wptherml

Suchanekf committed:
Merge pull request #71 from FoleyLab/new_release

New release

FoleyLab committed:
made a minor change

FoleyLab committed:
partial implementation of pv-stpv functionality

FoleyLab committed:
more updates to pv-stpv functionality

FoleyLab committed:
added some additional functionality to the pv-stpv stuff

FoleyLab committed:
updated readme!

FoleyLab committed:
Merge pull request #72 from FoleyLab/pioneer_fest

made a minor change

FoleyLab committed:
This really important - please include!

FoleyLab committed:
Merge pull request #74 from FoleyLab/escip

This really important - please include!

FoleyLab committed:
updated simple_tmm.py

FoleyLab committed:
Merge pull request #75 from FoleyLab/new_release

updated simple_tmm.py

FoleyLab committed:
updated simple_tmm

FoleyLab committed:
updated readme

FoleyLab committed:
updated the quickstart guide

FoleyLab committed:
Update quickstart.rst

FoleyLab committed:
Create wptherml_env.yml

added a yaml file for conda environment for wptherml

FoleyLab committed:
Merge branch 'main' of https://github.com/FoleyLab/wptherml

FoleyLab committed:
Update quickstart.rst

FoleyLab committed:
Update quickstart.rst

FoleyLab committed:
updated cooling demo

Co-Authored-By: lanetolley <113054686+lanetolley@users.noreply.github.com>

FoleyLab committed:
updated cooling demo with some data saved

Co-Authored-By: lanetolley <113054686+lanetolley@users.noreply.github.com>

FoleyLab committed:
created acoustic.py

FoleyLab committed:
added AcousticDriver

FoleyLab committed:
created simple example for AcousticDriver

Suchanekf committed:
Update acoustic.py

FoleyLab commented:
@Suchanekf thank you for this pull request!  I will keep this open until we have everything complete to have a functional compute_spectrum() method, but we can continue to make commits to this branch that will update the PR.

FoleyLab committed:
added some notes for S_mn

FoleyLab committed:
made S_mn an attribute

FoleyLab committed:
made compute_kx_ky_kz do it for a single value of k, theta, and phi

FoleyLab commented:
@Suchanekf I added a few notes for the S_mn calculation.  However, I think a little more thought may be required about how these different functions interact.  It may be advantageous to be able to compute kx, ky, and kz for a single value of k, theta, and phi rather than for an array (at least for a first pass because it is a little easier to understand), so I changed the compute_kx_ky_kz function to behave this way.  Now, you must pass it a value of k, theta, and phi and it will compute attributes self.kx_val, self.ky_val, and self.kz_val.  This way, it can be called within the nested for loops that are used for the S_mn calculation.

FoleyLab committed:
added cuboid.py for absorption / scattering of cuboids

FoleyLab committed:
added cuboiddriver - needs testing

Suchanekf commented:
> @Suchanekf I added a few notes for the S_mn calculation. However, I think a little more thought may be required about how these different functions interact. It may be advantageous to be able to compute kx, ky, and kz for a single value of k, theta, and phi rather than for an array (at least for a first pass because it is a little easier to understand), so I changed the compute_kx_ky_kz function to behave this way. Now, you must pass it a value of k, theta, and phi and it will compute attributes self.kx_val, self.ky_val, and self.kz_val. This way, it can be called within the nested for loops that are used for the S_mn calculation.

Thank you so much for the feedback. I will continue working on it.
